iceoryx_hoofs/test/moduletests/test_cxx_function.cpp

```diff
@@ -0,0 +1,630 @@
+}
+
+TEST_F(function_test, CopyAssignmentCopiesStoredFreeFunction)
+{
+    test_function f(freeFunction);
+    test_function sut(Functor(73));
+
+    Functor::resetCounts();
+    sut = f;
+
+    EXPECT_EQ(Functor::numDestroyed, 1U);
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    ASSERT_TRUE(sut.operator bool());
+    ASSERT_TRUE(f.operator bool());
+    EXPECT_EQ(sut(1), f(1));
+}
+
+TEST_F(function_test, MoveAssignmentMovesStoredFreeFunction)
+{
+    test_function f(freeFunction);
+    test_function sut(Functor(73));
+
+    Functor::resetCounts();
+    sut = std::move(f);
+
+    EXPECT_EQ(Functor::numDestroyed, 1U);
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    ASSERT_TRUE(sut.operator bool());
+    EXPECT_EQ(sut(1), freeFunction(1));
+    EXPECT_FALSE(f.operator bool());
+}
+
+TEST_F(function_test, CopiedNonCallableFunctionIsNotCallable)
+{
+    test_function f;
+    Functor::resetCounts();
+
+    test_function sut(f);
+
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    EXPECT_FALSE(sut.operator bool());
+    EXPECT_FALSE(f.operator bool());
+}
+
+TEST_F(function_test, MovedNonCallableFunctionIsNotCallable)
+{
+    test_function f;
+    Functor::resetCounts();
+
+    test_function sut(std::move(f));
+
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    EXPECT_FALSE(sut.operator bool());
+    EXPECT_FALSE(f.operator bool());
+}
+
+TEST_F(function_test, CopyAssignedNonCallableFunctionIsNotCallable)
+{
+    test_function f;
+    test_function sut(Functor(73));
+
+    Functor::resetCounts();
+    sut = f;
+
+    EXPECT_EQ(Functor::numDestroyed, 1U);
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    EXPECT_FALSE(sut.operator bool());
+    EXPECT_FALSE(f.operator bool());
+}
+
+TEST_F(function_test, MoveAssignedNonCallableFunctionIsNotCallable)
+{
+    test_function f;
+    test_function sut(Functor(73));
+
+    Functor::resetCounts();
+    sut = std::move(f);
+
+    EXPECT_EQ(Functor::numDestroyed, 1U);
+    EXPECT_EQ(Functor::numCopied, 0U);
+    EXPECT_EQ(Functor::numMoved, 0U);
+    EXPECT_FALSE(sut.operator bool());
+    EXPECT_FALSE(f.operator bool());
+}
+
+TEST_F(function_test, MemberSwapWorks)
+{
+    Functor f1(73);
+    Functor f2(37);
+    test_function sut1(f1);
+    test_function sut2(f2);
+
+    sut1.swap(sut2);
+
+    ASSERT_TRUE(sut1.operator bool());
+    EXPECT_EQ(sut1(1), f2(1));
+    ASSERT_TRUE(sut2.operator bool());
+    EXPECT_EQ(sut2(1), f1(1));
+}
+
+TEST_F(function_test, StaticSwapWorks)
+{
+    Functor f1(73);
+    Functor f2(37);
+    test_function sut1(f1);
+    test_function sut2(f2);
+
+    swap(sut1, sut2);
+
+    ASSERT_TRUE(sut1.operator bool());
+    EXPECT_EQ(sut1(1), f2(1));
+    ASSERT_TRUE(sut2.operator bool());
+    EXPECT_EQ(sut2(1), f1(1));
+}
+
+TEST_F(function_test, functorOfSizeSmallerThanStorageBytesCanBeStored)
```

Comment:
Nitpick: From here on all the test cases names are starting with a lowercase letter